### PR TITLE
refactor: remove internal TemplateRenderer use from TreeGrid

### PR DESCRIPTION
**Pull request by tomivirkki**
## Description

Replace `TemplateRenderer` with `LitRenderer` in `TreeGrid` 

Part of https://github.com/vaadin/flow-components/issues/4059

## Type of change

- Refactoring

### CODE REVIEW

File: vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/treegrid/it/TreeGridPreloadIT.java [hunk 1]
```diff
@@ -128,15 +128,13 @@ public void firstExpanded_scrollByViewportEstimate_shouldPreLoadDataForExpandedC
     }
 
     @Test
-    public void firstExpanded_reExpand_shouldPreLoadDataForExpandedChildren() {
```
Comment (tomivirkki) [inline vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/treegrid/it/TreeGridPreloadIT.java:131]: Seems that there's a difference if the expand item command originates from the server vs the client. 0 additional requests is better than 2 so repurposed these re-expand tests. There are other tests that validate that the content is correct after re-expanding.

File: vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/treegrid/TreeGrid.java [hunk 2]
```diff
@@ -555,14 +554,28 @@ public GridDataView<T> getGenericDataView() {
...
+        Column<T> column = addColumn(LitRenderer.<T> of(
+                "<vaadin-grid-tree-toggle @click=${onClick} .leaf=${!item.children} .expanded=${model.expanded} .level=${model.level}>"
+                        + "${item.name}</vaadin-grid-tree-toggle>")
                 .withProperty("children",
                         item -> getDataCommunicator().hasChildren(item))
                 .withProperty("name",
-                        value -> String.valueOf(valueProvider.apply(value))));
+                        value -> String.valueOf(valueProvider.apply(value)))
+                .withFunction("onClick", item -> {
+                    if (getDataCommunicator().hasChildren(item)) {
```
Comment (sissbruecker) [inline vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/treegrid/TreeGrid.java:570]: Should we add this condition to `HierarchyComponentRenderer` as well?
https://github.com/vaadin/flow-components/blob/158b50b10a3acd9a5b104d738c4015d284e0be39/vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/treegrid/HierarchyColumnComponentRenderer.java#L27-L33

File: vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/treegrid/TreeGrid.java [hunk 3]
```diff
@@ -555,14 +554,28 @@ public GridDataView<T> getGenericDataView() {
...
     public Column<T> addHierarchyColumn(ValueProvider<T, ?> valueProvider) {
-        Column<T> column = addColumn(TemplateRenderer
-                .<T> of("<vaadin-grid-tree-toggle "
-                        + "leaf='[[!item.children]]' expanded='{{expanded}}' level='[[level]]'>[[item.name]]"
-                        + "</vaadin-grid-tree-toggle>")
+
+        // The click listener needs to check if the event gets canceled (by
+        // vaadin-grid-tree-toggle) and only invoke the callback if it does.
+        // vaadin-grid-tree-toggle will cancel the event if the user clicks on
+        // a non-focusable element inside the toggle.
```
Comment (sissbruecker) [inline vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/treegrid/TreeGrid.java:561]: Looks like we are not doing that here though. Compared to `HierarchyColumnRenderer`:
https://github.com/vaadin/flow-components/blob/158b50b10a3acd9a5b104d738c4015d284e0be39/vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/treegrid/HierarchyColumnComponentRenderer.java#L41-L45

From that description it looks like it's not needed here, as we will always only have text node as content. Should we remove the comment then?

Comment (tomivirkki) [reply vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/treegrid/TreeGrid.java:561]: Oops. I first had the same check here but also noticed it's not needed so removed the logic. Apparently forgot to remove the comment. Done.

File: vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/treegrid/TreeGrid.java (hunk 2 above)
Comment (tomivirkki) [reply vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/treegrid/TreeGrid.java:570]: In `HierarchyComponentRenderer`s case, with the conditional click listener, the [check is done client-side](https://github.com/vaadin/web-components/blob/d6f871f2e694aa08748481d32ae9c7f6ba2e6b20/packages/grid/src/vaadin-grid-tree-toggle.js#L192)

File: vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/treegrid/TreeGrid.java (hunk 3 above)
Comment (tomivirkki) [reply vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/treegrid/TreeGrid.java:561]: That's odd. I pushed the commit but it doesn't show in GH yet

**Pull request merged**